### PR TITLE
fixed #21740: Implement sound change warning dialog for sound flags

### LINKED ISSUE #21740: Implement sound change warning dialog for sound flags
When I change a sound in the mixer for an part that has sound flags on it, I should be prompted with a warning dialog telling me that sound flags may be reset.

image

**Contents:**
Are you sure you want to change this sound?
Sound flags on this instrument may be reset, but stave text will remain. This action can’t be undone.
Don’t show this message again
[Cancel] [Change sound]

### PR DESCRIPTION
**Pull request by Eism**
Resolves: #21740

### CODE REVIEW

File: src/playback/view/internal/mixerchannelitem.cpp [hunk 1]
```diff
@@ -640,6 +652,39 @@ void MixerChannelItem::closeEditor(AbstractAudioResourceItem* item)
     item->setEditorUri(UriQuery());
 }
 
+bool MixerChannelItem::askAboutChangingSound()
+{
+    if (!notationPlayback()->hasSoundFlags({ m_instrumentTrackId })) {
+        return true;
+    }
+
+    if (!configuration()->needToShowChangeSoundWarning()) {
```
Comment (RomanPudashkin) [inline src/playback/view/internal/mixerchannelitem.cpp:661]: It is better (for performance) to check this before checking hasSoundFlags

**Comment by avvvvve** [comment]
@Eism A couple things:

- Change "stave text" to "staff text (for the US English default)
- If you click on the sound that is already selected in the sound selection context menus, the warning still shows. We can skip showing it in that case since you're not actually changing the sound

Otherwise looks good!

**Review by RomanPudashkin — APPROVED** [review]

**Comment by avvvvve** [comment]
@Eism Looks like something broke—the warning is now not appearing when it should. I changed Muse Sounds Trumpet to one of the new trumpet sounds and the sound flags disappeared with no warning.

**Comment by avvvvve** [comment]
@Eism 
1. I'm still seeing this issue:

     - If you click on the sound that is already selected in the sound selection context menus, the warning still shows. We can skip showing it in that case since you're not actually changing the sound

2. And just noticed that the icon should be the exclamation mark in a triangle (U+F3CE), not the question mark in the circle

**Comment by avvvvve** [comment]
Approved!

**Pull request merged**